llvm/docs/SYCLFrontEndDeviceCompilation.rst

```diff
@@ -0,0 +1,135 @@
+=================================
```

Comment:
`llvm/docs` doesn't seem like the right location for this documentation. `clang/docs` seems like a more appropriate location. There is already a `SYCLSupport.rst` file there that discusses address spaces (and that probably needs review). The `AMDGPUSupport.rst` file in that location will also need an update for SYCL (eventually).